src/liger_kernel/ops/experimental/gemm_split_k_fp8_e4m3.py

```diff
@@ -0,0 +1,302 @@
+# adapted from: https://github.com/pytorch-labs/applied-ai/blob/main/kernels/triton/inference/fp8/splitk_gemm_fp8.py
+# Valid Architectures: [SM_89, SM_90(a)]
+import torch
+import triton
+import triton.language as tl
+
+from liger_kernel.ops.utils import (
+    calculate_gemm_settings,
+    check_compute_capability_for_fp8,
+    ensure_contiguous,
+)
+
+
+@triton.jit
+def grouped_launch(
+    pid,
+    m,  # rows
+    n,  # cols
+    block_m: tl.constexpr,  # rows in a block
+    block_n: tl.constexpr,  # cols in a block
+    group_m: tl.constexpr,  # blocks in group along row dimension
+):
+    grid_m = tl.cdiv(m, block_m)
+    grid_n = tl.cdiv(n, block_n)
+    width = group_m * grid_n
+    group_id = pid // width
+    pid_in_group = pid % width
+    group_size = tl.minimum(grid_m - group_id * group_m, group_m)
+    pid_m = group_id * group_m + (pid_in_group % group_size)
+    pid_n = pid_in_group // group_size
+
+    return pid_m, pid_n
+
+
+@triton.jit
+def gemm_split_k_kernel_forward(
+    a_ptr,
+    b_ptr,
+    c_ptr,
+    stride_am,
+    stride_ak,
+    stride_bk,
+    stride_bn,
+    stride_cm,
+    stride_cn,
+    m,
+    n,
+    k,
+    block_m: tl.constexpr,
+    block_n: tl.constexpr,
+    block_k: tl.constexpr,
+    split_k: tl.constexpr,
+    group_m: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    pid_k = tl.program_id(1)
+    grid_k = tl.cdiv(k, block_k * split_k)
+
+    pid_m, pid_n = grouped_launch(pid, m, n, block_m, block_n, group_m)
+
+    offs_m = tl.multiple_of(pid_m * block_m + tl.arange(0, block_m), block_m)
+    offs_n = tl.multiple_of(pid_n * block_n + tl.arange(0, block_n), block_n)
+    offs_k = pid_k * block_k + tl.arange(0, block_k)
+
+    offs_am = tl.max_contiguous(tl.multiple_of(offs_m, block_m), block_m)
+    offs_bn = tl.max_contiguous(tl.multiple_of(offs_n, block_n), block_n)
+
+    a_ptrs = a_ptr + (offs_am[:, None] * stride_am + offs_k[None, :] * stride_ak)
+    b_ptrs = b_ptr + (offs_k[:, None] * stride_bk + offs_bn[None, :] * stride_bn)
+
+    acc = tl.zeros((block_m, block_n), dtype=tl.float32)
+
+    for k_ in range(0, grid_k, step=2):
+        k_remaining = k - k_ * (block_k * split_k)
+
+        mask_a = offs_k[None, :] < k_remaining
+        mask_b = offs_k[:, None] < k_remaining
+
+        a = tl.load(a_ptrs, mask=mask_a, other=0.0)
+        b = tl.load(b_ptrs, mask=mask_b, other=0.0)
+
+        # fp8 input dot product (supported types: [fp8e4nv, fp8e5, fp8e4b15])
+        acc = tl.dot(a, b, acc)
+
+        a_ptrs += block_k * split_k * stride_ak
+        b_ptrs += block_k * split_k * stride_bk
+
+    # cast to fp16 pre-store
+    acc = acc.to(tl.float16)
```

Comment:
Can we accumulate in fp16 directly to avoid the overhead of fp8>fp32>fp16 ?

Reply:
Yes, I can make that change and test.

Reply:
Re this:

I tried setting tl.dot output_dtype to tl.float16, but it doesn't seem to work. I don't think the tl.dot output dtype can be changed, so at least with this method, acc has to be float32.

Reply:
I think it has to use fp32 here but probably should cast to the given compute_type inferred from input rather than fixed as fp16 (as bf16 or fp32 might also be used)